mqtt_io/server.py

```diff
@@ -19,6 +19,7 @@
 from importlib import import_module
 from typing import Any, Dict, List, Optional, Tuple, Type, Union, overload
 
+import asyncio_mqtt  # type: ignore
```

Comment:
I've used an abstraction layer between the `server` module and the MQTT client(s), since I had to keep changing the clients. For this reason `asyncio_mqtt` should only be referenced in `mqtt_io/mqtt/asyncio_mqtt.py`.

This architecture is working, but still somewhat up in the air. I'm not sure if it's really necessary and it could definitely be implemented slightly better (not using protected member).

It'll probably just require making an Exception in `mqtt_io/mqtt/__init__.py` to mimic the one thrown by the asyncio-mqtt library.

Reply:
I've added a MQTTException now and the MQTTClient catches MqttErrors and re-raises them as MQTTException